LetsDoStuff.WebApi/Controllers/ActivityController.cs

```diff
@@ -42,14 +37,15 @@ public ActionResult<List<ActivityResponse>> GetActivities()
         [HttpGet("{id}")]
         public ActionResult<ActivityResponse> GetActivity(int id)
         {
-            var activity = _activityService.GetActivityById(id);
-
-            if (activity == null)
+            try
             {
-                return NotFound();
+                var activity = _activityService.GetActivityById(id);
+                return activity;
+            }
+            catch (ArgumentException ex)
+            {
+                return NotFound(ex.Message);
             }
-
-            return activity;
```

Comment:
Create a bug, describe the problem and open the request in the scope of the bug

Reply:
Sorry, I'm not sure what you are mean here. I changed this thing because this part of the code had the same problem that the example earlier: GetActivityById never returns null but in this case, it throws an ArgumentException if it doesn't find an activity by id. If you see a problem here, please give me more details.

Reply:
It's a bugfix for tl004, not for tl007